libnetdata/popen/popen.c

```diff
@@ -150,22 +189,22 @@ static inline int custom_popene(const char *command, volatile pid_t *pidptr, cha
     // Take the lock while we fork to ensure we don't race with SIGCHLD
     // delivery on a process which exits quickly.
     myp_add_lock();
-    if (!posix_spawn(&pid, "/bin/sh", &fa, &attr, spawn_argv, env)) {
+    if (!posix_spawn(&pid, command, &fa, &attr, (char * const*)spawn_argv, env)) {
         *pidptr = pid;
         myp_add_locked(pid);
-        debug(D_CHILDS, "Spawned command: '%s' on pid %d from parent pid %d.", command, pid, getpid());
+        debug(D_CHILDS, "Spawned command: \"%s\" on pid %d from parent pid %d.", command_to_be_logged, pid, getpid());
     } else {
         myp_add_unlock();
-        error("Failed to spawn command: '%s' from parent pid %d.", command, getpid());
-        if (flags & FLAG_CREATE_PIPE) {
+        error("Failed to spawn command: \"%s\" from parent pid %d.", command_to_be_logged, getpid());
+        if (flags & POPEN_FLAG_CREATE_PIPE) {
             fclose(fp);
         }
         ret = -1;
     }
-    if (flags & FLAG_CREATE_PIPE) {
+    if (flags & POPEN_FLAG_CREATE_PIPE) {
         close(pipefd[PIPE_WRITE]);
         if (0 == ret) // on success set FILE * pointer
-            *fpp = fp;
+            if(fpp) *fpp = fp;
```

Comment:
The way it was, this was leading to crashes if `POPEN_FLAG_CREATE_PIPE` was given but `fpp` was `NULL`.